filebeat/input/file/glob.go

```diff
@@ -0,0 +1,65 @@
+package file
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// globPattern detects the use of "**" and expands it to standard glob patterns up to a max depth
+func globPatterns(pattern string, doubleStarPatternDepth uint8) ([]string, error) {
+	if doubleStarPatternDepth == 0 {
+		return []string{pattern}, nil
+	}
+	var patterns []string
+	isAbs := filepath.IsAbs(pattern)
+	patternList := strings.Split(pattern, string(os.PathSeparator))
+	for i, dir := range patternList {
+		if len(patterns) > 0 {
+			if dir == "**" {
+				err := fmt.Sprintf("glob(%s) failed: cannot specify multiple ** within a pattern", pattern)
+				logp.Err(err)
```

Comment:
I don't recommend logging errors and returning them. Do one or the other. It's probably best to let the caller make the decision on how to handle the error (which might be to log it).